### DISCUSSION TITLE
Is Trivy Java DB a hard requirement for Jar analysis?

### DISCUSSION THREAD

**Discussion by Tehsmash** [aquasecurity/trivy#5266] [Q&A] [answered]
### Question

Today if Trivy finds a Jar file during image/filesystem SBOM analysis it is a hard requirement that the trivy-java-db be downloaded to the cache. Even when passing `--skip-java-db-update` it is enforced that the DB is downloaded at least once.

Looking at the code in go-dep-parser it appears that Jar files can ultimately be in three states:
- Has a valid pom.properties
- pom.properties is missing or invalid, but MANIFEST.MF has all the required properties
- pom.properties and MANIFEST.MF do not have enough information

It is only this third state that requires trivy-java-db to backfill / discover the missing information.

The trivy-java-db was developed for a reason, so we'd like to understand:

Is it a common scenario that Jars are malformed / missing information?

Is it possible to remove that requirement and document that if the trivy-java-db is not present/downloaded (always skipped) that Jars will be excluded from the SBOM unless accurate information can be discovered from the Jar itself? Will that result in a significant depreciation in SBOM accuracy for Java?

In our use case we're using Trivy for SBOM analysis in ephemeral nodes/containers, this results in the Java DB being downloaded for every analysis we want to run. Which is significant time/bandwidth overhead (465MiB today) when scanning many targets.

### Target

Filesystem

### Scanner

None

### Output Format

CycloneDX

### Mode

Standalone

### Operating System

Linux

### Version

```bash
./trivy --version
Version: 0.44.1
Java DB:
  Version: 1
  UpdatedAt: 2023-09-25 00:55:18.719389099 +0000 UTC
  NextUpdate: 2023-09-28 00:55:18.719388599 +0000 UTC
  DownloadedAt: 2023-09-25 13:59:38.617821 +0000 UTC
```

**Comment by knqyf263** [comment]
You can read the issue.
https://github.com/aquasecurity/trivy/issues/3427

**Comment by Tehsmash** [comment]
Hi @knqyf263  thanks for the reply, I've read through the issue that you linked and this doesn't answer the questions I have about why the java db / maven queries exist in the first place. From my understanding of that thread:

* Trviy used to call out to a remote maven API/DB to get additional information when it was missing from the Jar files
* trivy-java-db was created to remove the requirement on the remote APIs
* Comparing popular Jar files they use a mix of field names in the MANIFEST.MF to represent groupID

The last bullet was resolved by a PR which added some additional fields to the logic which populates the groupID.

That still leaves the main questions from above:

Is it a common scenario that Jars are malformed / missing information?

Is it possible to remove that requirement and document that if the trivy-java-db is not present/downloaded (always skipped) that Jars will be excluded from the SBOM unless accurate information can be discovered from the Jar itself? Will that result in a significant depreciation in SBOM accuracy for Java?

**Reply by knqyf263** [reply, marked as answer]
>Is it a common scenario that Jars are malformed / missing information?

Yes.

>Is it possible to remove that requirement

We have no such plans.

>Will that result in a significant depreciation in SBOM accuracy for Java?

Yes
